pontoon/base/views.py

```diff
@@ -98,6 +98,30 @@ def translate(request, locale, slug, part):
     })
 
 
+# LOCALE-AGNOSTIC VIEWS
+
+
+def locale_agnostic(request, slug, part):
+    """Language Agnostic view."""
+    user = request.user
+    project = get_object_or_404(Project.objects.available(), slug=slug)
+    
+    if user.is_authenticated():
+
+        locale = user.profile.custom_homepage:
+
+        if locale and project.locales.filter(code=locale).exists():
+            return redirect('pontoon.translate', slug=slug, locale=locale, part=part)
+        else:
```

Comment:
Hey @maiquynhtruong!
I feel that line will have to be verified with @adngdb and @mathjazz -> I assumed that this view will be responsible for validation of e.g. invalid locale names.
However, on the other side - if user types an invalid link/or something generates links to e.g. non-existing locales, we may want to monitor such errors to mitigate them in the future.

So, Welcome and thank you for your contribution!